Add cache when downloading XML files

vincent-psarga opened the pull request:
This PR adds some caching to hiptest-publisher, so it does not fetch the XML export from hiptest every time.

The default cache duration is 60 seconds, but that can be modified easily (```--xml-cache-validity=0``` for example).
Cached XML are stored in ```$HOME/.hiptest-publisher/cache```but this is also configurable (```--xml-cache-dir=/tmp``` for example)

cbliard commented:
Raw thoughts while I am testing it: 
 - `--xml-cache-validity` should be `--xml-cache-duration`, and maybe even `--cache-duration`: xml is an implementation detail that may change in the future
 - When using it for the second time, it hits the cache and is a lot faster. As it is not fetching data from Hiptest anymore on the second run, it should not display "Fetching data from HipTest" but rather something like "Reusing previously cached data from HipTest" or anything better.
 - In verbose mode, path to the cached files should be displayed

cbliard commented:
in verbose mode, it should also say something like "not using cached copy as it is xxx minutes/seconds old and cache-duration is set to xxx minutes/seconds

vincent-psarga commented:
- I've renamed the options so they don't relate anymore to "xml-"
- I've also added the verbose message when using the cache
- I've updated the message when using cache instead of downloading from Hiptest

I'm not sure about this:
> in verbose mode, it should also say something like "not using cached copy as it is xxx minutes/seconds old and cache-duration is set to xxx minutes/seconds

Normally, the cache files will be deleted automatically before checking them, so the message should never be shown.
